test/Libraries/DynamoPythonTests/PythonEvalTestsWithLibraries.cs

```diff
@@ -188,25 +187,35 @@ import clr
 from FFITarget import DummyCollection
 
 d = {'one': 1, 'two': 2, 'three': 3}
-# Python dict => DS Dictionary - Does not work in either engine
-# d2 = Dictionary.SetValueAtKeys(d, ['four'], [4])
 
-# Python dict => .NET IDictionary - Does not work in CPython
-d = DummyCollection.AcceptIDictionary(d)
-# .NET IDictionary => Python dict - Works in IronPython. Could not test in CPython due to previous bug
-d['five'] = 5
+# Python dict => .NET IDictionary
+untypedDictionary = DummyCollection.AcceptIDictionary(d)
+untypedDictionary['four'] = 4
 
-OUT = d
+# Python dict => .NET IDictionary<> - Does not work in IronPython
+typedDictionary = DummyCollection.AcceptDictionary(d)
```

Comment:
I understand that passing `d` to `AcceptDictionary` or IDictionary is the decoding part to convert it into the .NET dictionary but what about returning the .net dictionary from `AcceptDictionary`? Doesn't that need an encoding from the  returned .NET dictionary to the Python dictionary?

Reply:
Are the returned `untypedDictinoary` and `typedDictionary`  not Python dictionaries?

Reply:
It does trigger an encode operation but in this case we are not doing anything special. Python.NET will wrap .NET objects that it doesn't know how to encode with a generic wrapper. I haven't taken much of a look to their implementation, but they even support iterating like  this for example https://github.com/DynamoDS/Dynamo/blob/master/test/Libraries/DynamoPythonTests/PythonEvalTestsWithLibraries.cs#L77